client/src/components/Search/index.js

```diff
@@ -0,0 +1,144 @@
+            <Select
+              showSearch
+              placeholder="Select"
+              optionFilterProp="children"
+              onChange={value => this.setState({ country: value, city: '' })}
+              value={country}
+              filterOption={this.dropdownFilter}
+            >
+              {this.renderOptions(world.getCountryNames())}
+            </Select>
+          </div>
+          <div className="autocomplete-box">
+            <div>WHICH CITY?</div>
+            <Select
+              showSearch
+              placeholder="Select"
+              optionFilterProp="children"
+              onChange={value => {
+                this.setState({ city: value });
+              }}
+              value={city}
+              filterOption={this.dropdownFilter}
+            >
+              {country &&
+                world.getCities(country) !== undefined &&
+                this.renderOptions(world.getCities(country))}
+            </Select>
+          </div>
+        </div>
+
+        <div className="type-filter__container">
+          <Radio.Group
+            className="radio-group"
+            value={lookingFor}
+            buttonStyle="solid"
+          >
+            <p className="button-label">What Are You Looking For</p>
+            <Radio.Button
+              className="radio-button"
+              value="stay"
+              onClick={() => this.setState({ lookingFor: 'stay' })}
+            >
+              Stay
+            </Radio.Button>
+            <Radio.Button
+              className="radio-button"
+              value="eat"
+              onClick={() => this.setState({ lookingFor: 'eat' })}
+            >
+              Eat & drink
+            </Radio.Button>
+            <Radio.Button
+              className="radio-button"
+              value="shop"
+              onClick={() => this.setState({ lookingFor: 'shop' })}
+            >
+              Shop
+            </Radio.Button>
+          </Radio.Group>
+          <div className="type-filter__container__filter">
+            <p className="button-label">Filter</p>
+            <Button
+              className="keywords-btn"
+              onClick={() => {
+                this.setState(state => ({
+                  viewKeywords: !state.viewKeywords,
+                }));
+              }}
+            >
+              KeyWords
+            </Button>
+          </div>
+        </div>
+        <div className="form-action">
+          <Button className="search-btn" onClick={() => this.onSubmit()}>
+            Search
+          </Button>
+          <Button className="recommendation-btn" onClick={() => {}}>
+            + Add your recommendation
+          </Button>
+        </div>
+        {viewKeywords ? (
+          <Keywords
+            keyword={keyword}
+            setKeyword={this.setKeyword}
+            hideKeyword={this.hideKeyword}
+          />
+        ) : (
+          ''
+        )}
+      </form>
```

Comment:
the whole render function is quite long, consider splitting it into smaller functions